ateam_ui/install_deps.sh

```diff
@@ -6,10 +6,22 @@
 
 set -e
 
+if ! command -v npm; then
+  echo "ERROR: npm is not installed"
+
+  echo "attempting to install nodejs and npm"
+  sudo apt update -y
+  sudo apt install nodejs npm -y
+  if ! command -v npm; then
+    echo "failed to install npm"
+    exit 1
+  fi
+fi
+
 install_npm_package_if_missing () {
   local package_name=$1
   echo "Checking for npm package: $package_name"
-  if npm list --depth 1 --global $package_name > /dev/null 2>&1; then
+  if npm list --depth 1 --global $package_name > /dev/null; then
```

Comment:
don't suppress stderr if we get here. `set -e` is flagged above so the script will quit

Reply:
I'm curious if you hit a specific problem that motivated this change?
This command call is part of a conditional, so `set -e` should ignore it for the sake of its error checking. We're in fact counting on that behavior here, since this command returning 1 is our indicator that the package given needs to be installed.
I don't necessarily disagree with the change. I'm just not following the explanation.